Refactor to reasonableness

TehShrike opened the pull request.

TehShrike committed:
Test in node 8, whynot

TehShrike committed:
Refactor to be a pure function

Also changed the indentation to either "none" or "tabs" out of orneriness.

TehShrike committed:
Add package-lock whynot

TehShrike committed:
Deleting an apparently unused image?

TehShrike commented:
Oh hey, you should flip the Travis switch for this repo

TehShrike commented:
mm the readme still needs to be updated

ArtskydJ commented on test/index.js:
```diff
@@ -35,35 +40,33 @@ test('podcast', function(t) {
+					text: 'node.js',
+				}],
+			}],
 		}],
```
Why the trailing commas?

TehShrike replied:
Mostly all [this stuff](https://dnlserrano.github.io/2015/07/29/a-case-for-trailing-commas/) - personally, I'm to the point where I think I shouldn't have to type the commas at all, a newline should be enough inside of object literals, just like semiclons!

I added a rule to my eslint autoformatter for [dangling commas](https://eslint.org/docs/rules/comma-dangle).

ArtskydJ replied:
I don't mind them, I was just making sure it was intentional.

The argument that they make git diffs cleaner doesn't really apply in this case :stuck_out_tongue:

TehShrike replied:
lol, true

ArtskydJ commented:
lgtm

If you want to update the readme you can, otherwise I'll merge it and update the readme around the same time.

TehShrike committed:
Clean up the readme

Fixed some formatting oddities, added some details about child objects, updated to match the new syntax, removed some unnecessary text.
